internal/pkg/cli/selector/selector.go

```diff
@@ -69,13 +91,110 @@ func NewConfigSelect(prompt Prompter, store *config.Store) *ConfigSelect {
 
 // NewWorkspaceSelect returns a new selector that chooses applications and environments from the config store, but
 // services from the local workspace.
-func NewWorkspaceSelect(prompt Prompter, store *config.Store, ws *workspace.Workspace) *WorkspaceSelect {
+func NewWorkspaceSelect(prompt Prompter, store AppEnvLister, ws WsSvcLister) *WorkspaceSelect {
 	return &WorkspaceSelect{
 		Select:    NewSelect(prompt, store),
 		svcLister: ws,
 	}
 }
 
+// NewDeploySelect returns a new selector that chooses services and environments from the deploy store.
+func NewDeploySelect(prompt Prompter, configStore AppEnvLister, deployStore DeployStoreClient) *DeploySelect {
+	return &DeploySelect{
+		Select:         NewSelect(prompt, configStore),
+		deployStoreSvc: deployStore,
+	}
+}
+
+// GetServiceEnvironmentOpts sets up optional parameters for GetServiceEnvironmentOpts function.
+type GetServiceEnvironmentOpts func(*DeploySelect)
+
+// WithSvc sets up the svc name for DeploySelect.
+func WithSvc(svc string) GetServiceEnvironmentOpts {
+	return func(in *DeploySelect) {
+		in.svc = svc
+	}
+}
+
+// WithEnv sets up the env name for DeploySelect.
+func WithEnv(env string) GetServiceEnvironmentOpts {
+	return func(in *DeploySelect) {
+		in.env = env
+	}
+}
+
+// ServiceEnvironment fetches all deployed services and environments in an application and then
+// prompts the user to select one combination.
```

Comment:
I think this comment is a little confusing (at least for me). I think maybe we can have something like

```
DeployedService/ServiceEnvironment has the user select a deployed service. Callers can provide either a particular environment, a particular service to filter on, or both. 
```